team/bundles/org.eclipse.compare/compare/org/eclipse/compare/internal/CompareUIPlugin.java

```diff
@@ -1069,14 +1081,79 @@ public ViewerDescriptor[] findContentViewerDescriptor(Viewer oldViewer, Object i
 					result.addAll(list);
 			}
 			List<ViewerDescriptor> list = fContentMergeViewers.searchAll(ITypedElement.TEXT_TYPE);
-			if (list != null)
+			if (list != null) {
 				result.addAll(list);
-
-			return result.toArray(new ViewerDescriptor[0]);
+			}
 		}
+
+		ensureTextIsLast(result);
 		return result.isEmpty() ? null : result.toArray(new ViewerDescriptor[0]);
 	}
 
+	/**
+	 * Modifies given set to move "fallback" text descriptor to be the last one.
+	 * This is needed because we want more specific descriptors used first by default.
+	 */
+	private static void ensureTextIsLast(LinkedHashSet<ViewerDescriptor> result) {
+		if (result.size() > 1) {
+			ViewerDescriptor first = result.iterator().next();
+			if (TextMergeViewerCreator.class.getName().equals(first.getViewerClass())) {
+				result.remove(first);
+				result.add(first);
+			}
+		}
+	}
+
+	/**
+	 * @param fileName      possible file name for content in compare editor, may be
+	 *                      null
+	 * @param contentType   possible content type for content in compare editor, may
+	 *                      be null
+	 * @param firstIsEnough stop searching once first match is found
+	 * @return set of descriptors which could be found for given content type via
+	 *         "linked" editor
+	 */
+	Set<ViewerDescriptor> findEditorLinkedDescriptors(String fileName, IContentType contentType,
+			boolean firstIsEnough) {
+		if (fileName == null) {
+			if (contentType == null) {
+				contentType = fgContentTypeManager.findContentTypeFor(fileName);
```

Comment:
Seems little bit strange to me, since here `fileName` is `null`, isn't it ?

Reply:
Indeed, looks like it was meant if fileName != null above

Reply:
See https://github.com/eclipse-platform/eclipse.platform/pull/1459